test/UnitTest/Microsoft.AspNet.OData.Test.Shared/Formatter/Serialization/SelectExpandNodeTest.cs

```diff
@@ -214,12 +214,22 @@ public void BuildSelectExpandNode_ThrowsODataException_IfUnknownSelectItemPresen
         }
 
         [Fact]
-        public void ValidatePathIsSupported_ThrowsForUnsupportedPath()
+        public void ValidatePathIsSupported_ThrowsForUnsupportedPathForSelect()
         {
             ODataPath path = new ODataPath(new ValueSegment(previousType: null));
 
             ExceptionAssert.Throws<ODataException>(
-                () => SelectExpandNode.ValidatePathIsSupported(path),
+                () => SelectExpandNode.ValidatePathIsSupportedForSelect(path),
+                "A path within the select or expand query option is not supported.");
+        }
+
+        [Fact]
+        public void ValidatePathIsSupported_ThrowsForUnsupportedPathForExpand()
```

Comment:
>ValidatePathIsSupported_ThrowsForUnsupportedPathForExpand [](start = 20, length = 57)

is there positive test case?

Reply:
A lot of E2E positive tests. 

---
In reply to: [299586374](https://github.com/OData/WebApi/pull/1738#discussion_r299586374) [](ancestors = 299586374)